### PR TITLE
MueLu repartitioning: Allow use of graph partitioners

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/muelu 

## Description
Repartitioning using Zoltan2 only allowed the use of coordinate based repartitioning. This PR adds the option to use other, graph-based partitioners such as Scotch or ParMetis through Zoltan2. The default behavior is not changed.

## Motivation and Context
This allows repartitioning without having coordinates. I have not done any timing runs, but graph partitioning might be more expensive, so the setup cost might go up. On the other hand, graph partitioning minimizes the number of edge cuts, so we could potentially see faster solves.

### CODE REVIEW

**Comment by kddevin** [comment]
@cgcgcg I'm interested in seeing your timings when completed.  The graph partitioning will likely be more expensive and less scalable than geometric, but the setup-vs-solve tradeoffs may be interesting.  The XtraPuLP graph partitioning TPL may be more scalable; we can discuss enabling/trying XtraPuLP if you are interested.  Thanks.

**Comment by cgcgcg** [comment]
@kddevin Will do!

**Comment by aprokop** [comment]
@cgcgcg I'm quite cautious about how this is implemented, especially with the optional `Coordinates`. I'll try to review this by next week, would that work?

**Comment by cgcgcg** [comment]
@aprokop  Sure thing! We can talk about it in Portland..

File: packages/muelu/src/Rebalancing/MueLu_Zoltan2Interface_def.hpp [hunk 1]
```diff
@@ -95,7 +96,17 @@ namespace MueLu {
   void Zoltan2Interface<Scalar, LocalOrdinal, GlobalOrdinal, Node>::DeclareInput(Level& currentLevel) const {
     Input(currentLevel, "A");
     Input(currentLevel, "number of partitions");
-    Input(currentLevel, "Coordinates");
+    const ParameterList& pL = GetParameterList();
+    // We do this dance, because we don't want "ParameterList" to be marked as used.
+    // Is there a better way?
+    Teuchos::ParameterEntry entry = pL.getEntry("ParameterList");
```
Comment (cgcgcg) [inline packages/muelu/src/Rebalancing/MueLu_Zoltan2Interface_def.hpp]: @aprokop Following up on our earlier conversation: Using `"ParameterList"` here is actually the correct thing to do, as that's the name that is accessed down below. The translation from `"repartition: params"` to `"ParameterList"` happens in the ParameterListInterpreter.

**Comment by cgcgcg** [comment]
@aprokop Any chance you could review this?

**Pull request merged**